glue_jupyter/registries.py

```diff
@@ -0,0 +1,39 @@
+from glue.config import DictRegistry
+
+__all__ = ['viewer_registry', 'ViewerRegistry']
+
+
+class ViewerRegistry(DictRegistry):
+    """
+    Registry containing references to custom viewers.
```

Comment:
I wonder if this doesn't need to say 'custom' viewers as maybe we could actually add the regular viewers to it too so that they could also be created with the same API if desired?

Reply:
That's a good idea, but I have some questions/comments about the preferred way to do this.

1. Currently glupyter does not process plugin entry points from glue-jupyter (just from glue/glue-core). The built-in viewers are just imported in the functions in `app.py`. So in commit https://github.com/glue-viz/glue-jupyter/pull/307/commits/d214fe72cbdcaa40f46dae3d7b5a2aa794533cbe I have to import the viewers into the test functions before I can call them using the new (decorated) interface. The regular glue way is to add the viewers to the registry when their `setup` method runs from processing entry_points and all the default glue viewers have entries here. 
2. Since we have both bqplot and matplotlib viewers, the names for viewers are either ugly or default. That is, it probably makes sense to keep the bqplot viewers are default, so the name `scatter` means the bqplot scatter and if you wanted the matplotlib scatter we would call that `matplotlib-scatter` or something unintuitive like that.

Reply:
So, in summary -- this PR currently works fine for my current desire to add a new viewer to glue-jupyter via a separate plugin (this plugin calls glue.entry_points to add the viewer to the registry). This PR does NOT YET add all the existing viewers to the registry to enable them to be called via:

`app.new_data_viewer('table')`

although it could. My main hesitation (apart from the naming question I reference above) is that this `new_data_viewer` interface does not support any configuration options at creation time (e.g. `x = some_attribute`). I could extend it to do so -- we basically try to create state objects from all kwargs except `show` and `data` -- but that is a more significant change to the API if any other code is already calling `new_data_viewer`. This new method for creating viewers will always be less desirable from a notebook because you don't have the nice custom docstring that the dedicated viewer creation functions (e.g. `histogram_1d`) have for helping the user know what parameters are available at viewer creation time so perhaps it's not worth the effort?